File: code/backend/java/core/src/main/java/games/paths/core/model/story/CreatorInfo.java
package games.paths.core.model.story;

/**
 * CreatorInfo - Domain model for a content creator profile.
 * Used to expose creator details for cards, texts, and stories.
 * Immutable record (added in Step 16 for the content detail APIs).
 */
public record CreatorInfo(
        String uuid,
        String name,
        String link,
        String url,
        String urlImage,
        String urlEmote,
        String urlInstagram) {
}

File: code/backend/python/app/core/models/story/creator_info.py
from dataclasses import dataclass

@dataclass
class CreatorInfo:
    uuid: str
    name: str | None = None
    link: str | None = None
    url: str | None = None
    urlImage: str | None = None
    urlEmote: str | None = None
    urlInstagram: str | None = None
